### PR TITLE
Dependency updates

### PR DESCRIPTION
**Pull request by daira**
Update dependencies and audits, and bump MSRV to 1.66.

### CODE REVIEW

**Comment by daira** [comment]
Audit updates are not complete yet.

File: Cargo.lock [hunk 1]
```diff
@@ -1536,25 +1507,25 @@ dependencies = [
...
 source = "registry+https://github.com/rust-lang/crates.io-index"
-checksum = "4c42a9226546d68acdd9c0a280d17ce19bfe27a46bf68784e4066115788d008e"
+checksum = "1e401f977ab385c9e4e3ab30627d6f26d00e2c73eef317493c4ec6d468726cf8"
 dependencies = [
  "cfg-if",
  "libc",
  "redox_syscall",
  "smallvec",
- "windows-targets",
+ "windows-targets 0.52.5",
```
Comment (daira) [inline Cargo.lock:1528]: The duplication between 0.52.0 and 0.52.5 could potentially be avoided I guess. Not sure why this is requiring 0.52.5.

File: Cargo.toml [hunk 2]
```diff
@@ -96,8 +96,8 @@ rayon = "1.5"
 
 # Protobuf and gRPC
 prost = "0.12"
-tonic = { version = "0.10", default-features = false }
-tonic-build = { version = "0.10", default-features = false }
+tonic = { version = "0.11", default-features = false }
+tonic-build = { version = "0.11", default-features = false }
```
Comment (str4d) [inline Cargo.toml:100]: [`tonic 0.11` bumps MSRV to 1.70](https://github.com/hyperium/tonic/blob/master/CHANGELOG.md#0110-2024-02-08). That also happens to be the next MSRV I wanted to target for `zcash_primitives` and its downstream crates (I'm more cautious about bumping MSRV for crates like `pasta_curves`, but for the crates in this workspace 1.70 should be fine). We were blocked from doing so by `zcashd` being stuck on 1.69, but that was finally fixed last month in https://github.com/zcash/zcash/pull/6879.

I note that in the first commit you avoided certain other dependency updates that would require 1.70; we can upgrade those now. Given that `tonic 0.11` only forces the case for `zcash_client_backend` with the relevant feature flag enabled, I'm fine with merging this PR as-is, and then landing a follow-up PR that uniformly bumps MSRV to 1.70.

File: components/equihash/CHANGELOG.md [hunk 3]
```diff
@@ -6,6 +6,8 @@ and this library adheres to Rust's notion of
 [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Changed
+- MSRV is now 1.66.0.
```
Comment (str4d) [inline components/equihash/CHANGELOG.md:10]: Most of the `components/` crates use their own MSRVs as they are much more stable and rarely updated (note they do not use `rust-version.workspace = true`, so the top-level workspace MSRV change did not alter them). Revert these changes.
```suggestion
```

File: components/f4jumble/CHANGELOG.md [hunk 4]
```diff
@@ -6,6 +6,8 @@ and this library adheres to Rust's notion of
 [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Changed
+- MSRV is now 1.66.0.
```
Comment (str4d) [inline components/f4jumble/CHANGELOG.md:10]:
```suggestion
```

File: components/zcash_address/CHANGELOG.md [hunk 5]
```diff
@@ -6,14 +6,16 @@ and this library adheres to Rust's notion of
...
 ### Added
 - `zcash_address::ZcashAddress::{can_receive_memo, can_receive_as, matches_receiver}`
 - `zcash_address::unified::Address::{can_receive_memo, has_receiver_of_type, contains_receiver}`
 - Module `zcash_address::testing` under the `test-dependencies` feature.
 - Module `zcash_address::unified::address::testing` under the 
   `test-dependencies` feature.
 
+### Changed
+- MSRV is now 1.66.0.
+
```
Comment (str4d) [inline components/zcash_address/CHANGELOG.md:18]:
```suggestion
```

File: components/zcash_encoding/CHANGELOG.md [hunk 6]
```diff
@@ -6,6 +6,8 @@ and this library adheres to Rust's notion of
 [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Changed
+- MSRV is now 1.66.0.
```
Comment (str4d) [inline components/zcash_encoding/CHANGELOG.md:10]:
```suggestion
```

File: components/zcash_protocol/CHANGELOG.md [hunk 7]
```diff
@@ -9,6 +9,9 @@ and this library adheres to Rust's notion of
 ### Added
 - `zcash_protocol::PoolType::{TRANSPARENT, SAPLING, ORCHARD}`
 
+### Changed
+- MSRV is now 1.66.0.
```
Comment (str4d) [inline components/zcash_protocol/CHANGELOG.md:13]: This component crate is an exception, as it was recently extracted from `zcash_primitives` and thus shares a previous MSRV (and is less of a component crate and more of a crate we will regularly be updating, so a synced MSRV makes sense).

File: components/zip321/CHANGELOG.md [hunk 8]
```diff
@@ -32,3 +32,4 @@ The entries below are relative to the `zcash_client_backend` crate as of
   - `zip321::parse::{lead_addr, zcashparam}`
 - `zip321::Param::Memo` now boxes its argument.
 - `zip321::Param::Addr` now wraps a `zcash_address::ZcashAddress`
+- MSRV is now 1.66.0.
```
Comment (str4d) [inline components/zip321/CHANGELOG.md:35]: This crate was also recently extracted from another crate with a higher MSRV than other component crates, and currently is synced with the workspace MSRV. It doesn't necessarily need to be though as I expect it to be a bit more stable, but fine to keep it synced for now.

File: zcash_client_backend/CHANGELOG.md [hunk 9]
```diff
@@ -29,6 +29,7 @@ and this library adheres to Rust's notion of
   wrapping protocol-address types, the `Recipient` type now wraps a
   `zcash_address::ZcashAddress`. This simplifies the process of tracking the
   original address to which value was sent.
+- MSRV is now 1.66.0.
```
Comment (str4d) [inline zcash_client_backend/CHANGELOG.md:32]: I don't actually know if `tonic-build` also had its MSRV bumped (they are developed in the same workspace, but neither have explicit `rust-version`s), but given that CI is passing I think that means it currently isn't using post-1.66 features. So the only thing we should note here is that the `lightwalletd-tonic` feature has MSRV 1.70.0 (until we change workspace MSRV to 1.70 in a subsequent PR):
```suggestion
- MSRV is now 1.66.0 (except for the `lightwalletd-tonic` feature flag which
  requires 1.70.0).
```

File: supply-chain/audits.toml [hunk 10]
```diff
@@ -7,7 +7,354 @@ description = "The cryptographic code in this crate has been reviewed for correc
...
+
+[[audits.rustls]]
+who = "Daira-Emma Hopwood <daira@jacaranda.org>"
+criteria = "safe-to-deploy"
+delta = "0.21.8 -> 0.21.12"
+notes = """
+A comment in get_sni_extension asks whether the behaviour of parsing an IPv4 or IPv6 address
+in a host_name field of a server_name extension, but then ignoring the extension (because
+'Literal IPv4 and IPv6 addresses are not permitted in \"HostName\"'), as the server, is
+compliant with RFC 6066. As an original author of RFC 3546 which has very similar wording,
```
Comment (str4d) [inline supply-chain/audits.toml:204]: Neat!

**Review by str4d — APPROVED** [review]
utACK after changelog fixes.

**Review by str4d — APPROVED** [review]
Re-utACK 851e20347f0ea304f3083830ad65bc3fa389365b

**Comment by daira** [comment]
Post-hoc ACK 851e20347f0ea304f3083830ad65bc3fa389365b.
